app/src/main/res/layout/activity_login.xml

```diff
@@ -0,0 +1,177 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/white">
+
+    <android.support.constraint.Guideline
+        android:id="@+id/guide_line_left"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintGuide_percent="0.1"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <android.support.constraint.Guideline
+        android:id="@+id/guide_line_right"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintGuide_percent="0.9"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <android.support.constraint.Guideline
+        android:id="@+id/guild_line_top"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:orientation="horizontal"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintGuide_percent="0.1"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <EditText
+        android:id="@+id/edit_text_user_name"
+        style="@style/EditTextCustom"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:autofillHints="@string/app_name"
+        android:hint="@string/login_email_hint"
+        android:inputType="textPersonName"
+        android:maxLines="1"
+        app:layout_constraintEnd_toStartOf="@+id/guide_line_right"
+        app:layout_constraintStart_toEndOf="@+id/guide_line_left"
+        app:layout_constraintTop_toBottomOf="@+id/guild_line_top"
+        tools:ignore="LabelFor"
+        tools:targetApi="o" />
+
+    <EditText
+        android:id="@+id/edit_text_password"
+        style="@style/EditTextCustom"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dimen_8dp"
+        android:autofillHints="@string/app_name"
+        android:hint="@string/login_password_hint"
+        android:inputType="textPassword"
+        android:maxLines="1"
+        app:layout_constraintEnd_toStartOf="@+id/guide_line_right"
+        app:layout_constraintStart_toStartOf="@+id/guide_line_left"
+        app:layout_constraintTop_toBottomOf="@+id/edit_text_user_name"
+        tools:ignore="LabelFor"
+        tools:targetApi="o" />
```

Comment:
https://developer.android.com/studio/write/tool-attributes#toolstargetapi
em đọc thêm về targetApi nhé.
hiện a đang thấy e để targetApi là = 'o', như vậy thì các máy có sdk < 'o' sẽ k sử dụng được
[info]This tells the tools that you believe this element (and any children) will be used only on the specified API level or higher. This stops lint from warning you if that element or its attributes are not available on the API level you specify as your minSdkVersion.[/info]